src/v/net/server.cc

```diff
@@ -151,10 +151,17 @@ void server::print_exceptional_future(
               ctx,
               address,
               ex);
+        } else if (is_message_error(ex)) {
```

Comment:
Pre-existing issue:

This code path is
```
if () {
  if () {
  } else {
  }
}else {
}
```

while it could be an if-elseif chain.
Given that you are already here, would it be worth the hassle of fixing it?